src/test/java/edu/cornell/kfs/module/purap/service/impl/JaggaerRoleServiceImplTest.java

```diff
@@ -0,0 +1,235 @@
+    void testisDefaultJaggaerRole(String role, boolean expected) {
+        boolean actual = jaggaerRoleService.isDefaultJaggaerRole(role);
+        assertEquals(expected, actual);
+    }
+    
+    private static Stream<Arguments> provideRolesForIsDefaultJaggaerRoleTest() {
+        return Stream.of(
+          Arguments.of(JaggaerConstants.JAGGAER_ROLE_NAME_VIEW_ONLY, true),
+          Arguments.of(CUPurapConstants.SCIQUEST_ROLE_SHOPPER, false),
+          Arguments.of(StringUtils.EMPTY, false),
+          Arguments.of(null, false)
+        );
+    }
+    
+    @ParameterizedTest
+    @MethodSource("provideRolesForGetEshopPreAuthValueTest")
+    void testGetEshopPreAuthValue(boolean isBuyer, String expectedRole) {
+        jaggaerRoleService.setPermissionService(buildMockPermissionService(isBuyer, false, false, false));
+        String actual = jaggaerRoleService.getEshopPreAuthValue(user.getPrincipalId());
+        assertEquals(expectedRole, actual);
+    }
+    
+    private static Stream<Arguments> provideRolesForGetEshopPreAuthValueTest() {
+        return Stream.of(
+          Arguments.of(true, CUPurapConstants.SCIQUEST_ROLE_BUYER),
+          Arguments.of(false, CUPurapConstants.SCIQUEST_ROLE_SHOPPER)
+        );
+    }
+    
+    @ParameterizedTest
+    @MethodSource("provideRolesForGetEshopViewValueTest")
+    void testGetEshopViewValue(boolean isOffice, boolean isLab, boolean isFacilities, String expectedRole) {
+        jaggaerRoleService.setPermissionService(buildMockPermissionService(true, isOffice, isLab, isFacilities));
+        String actual = jaggaerRoleService.getEshopViewValue(user.getPrincipalId());
+        assertEquals(expectedRole, actual);
+    }
+    
+    private static Stream<Arguments> provideRolesForGetEshopViewValueTest() {
+        return Stream.of(
+          Arguments.of(true, false, false, CUPurapConstants.SCIQUEST_ROLE_OFFICE),
+          Arguments.of(false, true, false, CUPurapConstants.SCIQUEST_ROLE_LAB),
+          Arguments.of(false, false, true, CUPurapConstants.SCIQUEST_ROLE_FACILITIES),
+          Arguments.of(false, false, false, CUPurapConstants.SCIQUEST_ROLE_UNRESTRICTED)
+        );
+    }
+    
+    
+    @ParameterizedTest
+    @MethodSource("provideforTestGetJaggaerRoles")
+    void testGetJaggaerRoles(boolean isBuyer, boolean isOffice, boolean isLab, boolean isFacilities, List<String> expectedRoles) {
+        jaggaerRoleService.setJaggaerRoleLinkMappingService(buildMockJaggaerRoleLinkMappingService());
+        jaggaerRoleService.setPermissionService(buildMockPermissionService(isBuyer, isOffice, isLab, isFacilities));
+        List<String> actualRoles = jaggaerRoleService.getJaggaerRoles(user, JaggaerRoleSet.ADMINISTRATOR);
+        assertEquals(expectedRoles.size(), actualRoles.size());
+        for (String role : actualRoles) {
+            assertTrue(expectedRoles.contains(role));
+        }
+    }
+    
+    private static Stream<Arguments> provideforTestGetJaggaerRoles() {
```

Comment:
Minor typo in camel-casing of the method name.

Reply:
switching to provideForTestGetJaggaerRoles